### DISCUSSION TITLE
Add support for default values in the Repeater interface

### DISCUSSION THREAD

**Discussion by sybena** [directus/directus#24687] [Feature Requests]
### Describe the Bug

The current implementation of default values for Repeater fields in Directus has a significant limitation: the default value applies only to the initial state of the entire Repeater field (e.g., the first entry). However, when a new item is added to the Repeater, all fields within that new entry are empty, as the default value cannot be defined for the individual fields inside the Repeater structure.

**Current behavior:**

- The default value initializes the Repeater field with one prefilled entry, but does not affect any subsequent entries.
- Users must manually fill in all fields for every new entry, even if consistent default values are required for each added item.

**Proposed enhancement:**

- Allow specifying default values for the individual fields inside the Repeater structure.
- These defaults would be applied automatically whenever a new entry is created within the Repeater.

This enhancement would significantly improve usability and streamline workflows for users who rely on Repeater fields with recurring structures.

### To Reproduce

**1. Create a Repeater field:**

Open any collection in Directus and add a new field of type Repeater.

**2. Configure the interface:**

In the interface settings for the Repeater, add several subfields (e.g., a text field, a number field).

**3. Search for default value options:**

Look in the configuration settings of the subfields for an option to define default values for these fields.

### Directus Version

v11.2.1

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by BillyBouman-2B-IT** [comment]
This would enhance the default value functionality for the repeater fields indeed.

**Comment by rijkvanzanten** [comment]
The `fields` information is already passed to the `v-form` rendering the fields in the interface drawer, so adding it in the options here should do the trick:

https://github.com/directus/directus/blob/f718059d6cc0ae09fff444de5f0762a9cfaeef4b/app/src/interfaces/list/options.vue

That being said, right now it's mapped to only render the `meta` properties of the fields, whereas the default value is in `schema`. This file will have to be refactored slightly to correctly manage both meta and schema for the various fields. PR is more than welcome 🙂
